plugin/file/metrics.go

```diff
@@ -0,0 +1,56 @@
+package file
+
+import (
+	"github.com/coredns/coredns/plugin"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const subsystem = "secondary"
+
+// Metrics for the secondary plugin
+var (
+	UpdatedSeconds = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: plugin.Namespace,
+		Subsystem: subsystem,
+		Name:      "update_duration_seconds",
+		Buckets:   plugin.TimeBuckets,
+		Help:      "Histogram of the duration of an update",
+	}, []string{"zone"})
+
+	UpdatedSerial = prometheus.NewGaugeVec(prometheus.GaugeOpts{
```

Comment:
I've recently come to the realisation that gauges are bad, like really bad for monitoring. And it's much better to make this a counter and use epoch timestamp to convey this information. Just like process_start_time_second which just stamps things with the unix epoch when it happens.

Reply:
+1, in general gauges can be bad for monitoring.

However, the zone serial is an integer, and could be useful for comparison. For example, if you have multiple secondaries serving a zone, you could compare something like this:

`coredns_secondary_serial != ignoring (instance) max without (instance) (coredns_secondary_serial)`

This would let you know if one of several secondaries isn't getting updated.